applications/AdjointFluidApplication/custom_schemes/adjoint_bossak_scheme.h

```diff
@@ -662,6 +684,22 @@ class AdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace>
                 NUMBER_OF_NEIGHBOUR_ELEMENTS.Zero();
     }
 
+    void CalculateAdjointEnergy(ModelPart& rModelPart)
+    {
+        double total_energy = 0.0;
+        for (auto it = rModelPart.NodesBegin(); it != rModelPart.NodesEnd(); ++it) 
+        {
+            double v_x = it->FastGetSolutionStepValue(ADJOINT_FLUID_VECTOR_1_X, 0);
+            double v_y = it->FastGetSolutionStepValue(ADJOINT_FLUID_VECTOR_1_Y, 0);
```

Comment:
performance comment: the following would be better

       const array_1d<double,3>& v = it->FastGetSolutionStepValue(ADJOINT_FLUID_VECTOR);
      total_energy += inner_prod(v,v) + p*p;

the point is using one single vector access to the database instead of one per each component